coins/monero/tests/scan.rs

```diff
@@ -0,0 +1,160 @@
+
+    // get the tx and confirm receipt
+    tx = rpc.get_transaction(tx.hash()).await.unwrap();
+    let output = addr_scanner.scan_transaction(&tx).not_locked().swap_remove(0);
+    assert_eq!(output.commitment().amount, amount);
+    if addr.payment_id.is_some() {
+      assert_eq!(output.metadata.payment_id, addr.payment_id.unwrap());
+    }
+
+    // pick another input to spend for the  next address
+    let block = rpc.get_block(start + i).await.unwrap();
+    input = miner_scanner
+      .scan(&rpc, &block)
+      .await
+      .unwrap()
+      .swap_remove(0)
+      .ignore_timelock()
+      .swap_remove(0);
+  }
+}
+
+async_sequential!(
+  async fn scan_all_standard_addresses() {
+    // scanner
+    let spend_pub = &Scalar::from_bits(ADDR_SPEND) * &ED25519_BASEPOINT_TABLE;
+    let mut scanner = Scanner::from_view(
+      ViewPair::new(spend_pub, Zeroizing::new(Scalar::from_bits(ADDR_VIEW))),
+      Network::Mainnet,
+      Some(HashSet::new()),
+    );
+
+    // generate all types of addresses
+    let payment_ids = vec![
+      [46, 48, 134, 34, 245, 148, 243, 195],
+      [153, 176, 98, 204, 151, 27, 197, 168],
+      [88, 37, 149, 111, 171, 108, 120, 181],
+    ];
+    let mut addresses = vec![];
+    // standard versions
+    addresses.push(AddressInfo { address: scanner.address().to_string(), payment_id: None });
+    addresses
+      .push(AddressInfo { address: scanner.subaddress((0, 1)).to_string(), payment_id: None });
+    addresses.push(AddressInfo {
+      address: scanner.integrated_address(payment_ids[0]).to_string(),
+      payment_id: Some(payment_ids[0]),
+    });
+    // featured versions
+    addresses.push(AddressInfo {
+      address: scanner.featured_address(Some((0, 2)), Some(payment_ids[1]), false).to_string(),
+      payment_id: Some(payment_ids[1]),
+    });
+    addresses.push(AddressInfo {
+      address: scanner.featured_address(None, None, false).to_string(),
+      payment_id: None,
+    });
+    addresses.push(AddressInfo {
+      address: scanner.featured_address(Some((0, 3)), None, false).to_string(),
+      payment_id: None,
+    });
+    addresses.push(AddressInfo {
+      address: scanner.featured_address(None, Some(payment_ids[2]), false).to_string(),
+      payment_id: Some(payment_ids[2]),
+    });
+
+    // send to &  test receive from addresses
+    scan_incomings(&mut scanner, &addresses).await;
+  }
+);
+
+async_sequential!(
+  async fn scan_all_guaranteed_addresses() {
+    // scanner
+    let spend_pub = &Scalar::from_bits(ADDR_SPEND) * &ED25519_BASEPOINT_TABLE;
+    let mut scanner = Scanner::from_view(
+      ViewPair::new(spend_pub, Zeroizing::new(Scalar::from_bits(ADDR_VIEW))),
+      Network::Mainnet,
+      None,
+    );
+
+    // generate all types of addresses
+    let payment_ids =
+      vec![[88, 37, 149, 111, 171, 108, 120, 181], [125, 69, 155, 152, 140, 160, 157, 186]];
+    let mut addresses = vec![];
```

Comment:
You can write this entire section using `vec![]`.
```
vec![
  // comment
  AddressInfo {},
  ...
]
```
It'd remove the pushes and offer consistent formatting.